web/app.js

```diff
@@ -1851,6 +1857,11 @@ const PDFViewerApplication = {
     this.pdfPresentationMode?.request();
   },
 
+  requestFreeTextMode() {
+    AnnotationEditorLayerBuilder.setEditorType("FreeText");
+    this.appConfig.toolbar.freeTextModeButton.classList.toggle("toggled");
+  },
+
```

Comment:
Please handle this in the `BaseViewer` instead, again similar to how the scroll/spread-modes are implemented.

Basically, let's add a `annotationEditorMode` getter/setter-pair in the BaseViewer that takes care of updating the editor-mode and then dispatches a `annotationeditormodechanged` event that the Toolbar listens to and updates the buttons correctly.